lib/doctest.mjs

```diff
@@ -0,0 +1,67 @@
+import {readFile, writeFile, unlink} from 'fs';
+import {resolve} from 'path';
+import {promisify} from 'util';
+
+import common from './common';
+import doctest from './doctest.js';
+
+export default async function(path, options) {
+  if (options.module === 'esm') {
+    if (options.type != null && options.type !== 'esm') {
+      throw new Error (
+        'EcmaScript modules only work with the EcmaScript module type'
+      );
```

Comment:
I think this message should mention `--module esm`.

Reply:
I'm not sure - it may not have been used through the CLI.

Reply:
This error message has changed altogether now.